frontend/packages/rhoas-plugin/src/components/service-list/ServiceListPage.tsx

```diff
@@ -18,36 +26,51 @@ import {
 } from '../../utils/resourceCreators';
 import { KafkaRequest } from '../../utils/rhoas-types';
 import {
-  isResourceStatusSuccessfull,
-  isAcccesTokenSecretValid,
+  isResourceStatusSuccessful,
+  isAccessTokenSecretValid,
   getFinishedCondition,
 } from '../../utils/conditionHandler';
-import { ServicesErrorState } from '../states/ServicesErrorState';
+import { useTranslation } from 'react-i18next';
+import { ServicesEmptyState } from '../states';
+
+type ConnectionErrorProps = {
+  title: string;
+  message: string;
+  action: () => void;
+  actionLabel: string;
+};
 
 const ServiceListPage: React.FC = () => {
   const [currentNamespace] = useActiveNamespace();
   const [selectedKafka, setSelectedKafka] = React.useState<number>();
   const [currentKafkaConnections, setCurrentKafkaConnections] = React.useState<string[]>();
-  const { t } = useTranslation();
-  const [kafkaCreateError, setKafkaCreateError] = React.useState<string>();
-  const [kafkaListError, setKafkaListError] = React.useState<string>();
   const [isSubmitting, setSubmitting] = React.useState<boolean>(false);
+  const [connectionError, setConnectionError] = React.useState<ConnectionErrorProps>();
+
+  const { t } = useTranslation();
 
   React.useEffect(() => {
     const createKafkaRequestFlow = async () => {
       try {
         await createCloudServicesRequestIfNeeded(currentNamespace);
-
         const currentKafka = await listOfCurrentKafkaConnectionsById(currentNamespace);
         if (currentKafka) {
           setCurrentKafkaConnections(currentKafka);
         }
       } catch (error) {
-        setKafkaListError(error);
+        const connectionErrorObj = {
+          title: t('rhoas-plugin~Could not fetch services'),
+          message: t('rhoas-plugin~Failed to load list of services', {
+            error,
+          }),
+          action: () => history.push(`/catalog/ns/${currentNamespace}?catalogType=managedservices`),
```

Comment:
[Future] hardcoded catalog type

Reply:
Actually there will be the same result as this
```suggestion
          action: () => history.goBack(),
```

Reply:
@wtrocki but what if the user has come from the topology page? it would go back to the topology page instead of the services list right ?

Reply:
This is history wrapper for react router. The only way to get to that page is thru catalog.

I might be wrong so do not take me for granted.

Reply:
so if the user goes catalog --> service list --> topology --> back to service list ... can we guarantee that the request to the get the list of kafkas will always be true after the user visits the topology page? like is that data cached ?

otherwise i think we need to specify the direct path back to the catalog